examples/ex03_proc/ProcRTL.py

```diff
@@ -53,112 +53,110 @@ def construct( s ):
-    connect_pairs(
-      m.in_.en,  s.imemresp_q.deq.en,
-      m.in_.rdy, s.imemresp_q.deq.rdy,
-      m.in_.ret, s.imemresp_q.deq.ret.data,
-    )
+    m.in_.en  //= s.imemresp_q.deq.en
+    m.in_.rdy //= s.imemresp_q.deq.rdy
+    m.in_.ret //= s.imemresp_q.deq.ret.data
+
+    # connect all the queues
+
+    s.imemreq_q.deq   //= s.imem.req
+    s.imemresp_q.enq  //= s.imem.resp
+    s.dmemresp_q.enq  //= s.dmem.resp
+    s.mngr2proc_q.enq //= s.mngr2proc
+    s.xcelresp_q.enq  //= s.xcel.resp
 
     # Control
 
-    s.ctrl  = ProcCtrl()(
-
-      # imem port
-      imemresp_drop = s.imemresp_drop.drop,
-      imemreq_en    = s.imemreq_q.enq.en,
-      imemreq_rdy   = s.imemreq_q.enq.rdy,
-      imemresp_en   = s.imemresp_drop.out.en,
-      imemresp_rdy  = s.imemresp_drop.out.rdy,
-
-      # dmem port
-      dmemreq_en    = s.dmem.req.en,
-      dmemreq_rdy   = s.dmem.req.rdy,
-      dmemreq_type  = s.dmem.req.msg.type_,
-      dmemresp_en   = s.dmemresp_q.deq.en,
-      dmemresp_rdy  = s.dmemresp_q.deq.rdy,
-
-      # xcel port
-      xcelreq_en    = s.xcel.req.en,
-      xcelreq_rdy   = s.xcel.req.rdy,
-      xcelresp_en   = s.xcelresp_q.deq.en,
-      xcelresp_rdy  = s.xcelresp_q.deq.rdy,
-
-      # proc2mngr and mngr2proc
-      proc2mngr_en  = s.proc2mngr.en,
-      proc2mngr_rdy = s.proc2mngr.rdy,
-      mngr2proc_en  = s.mngr2proc_q.deq.en,
-      mngr2proc_rdy = s.mngr2proc_q.deq.rdy,
-
-      # commit inst for counting
-      commit_inst = s.commit_inst
-    )
+    s.ctrl = m = ProcCtrl()
 
-    # Dpath
+    # imem port
+    m.imemresp_drop //= s.imemresp_drop.drop
+    m.imemreq_en    //= s.imemreq_q.enq.en
+    m.imemreq_rdy   //= s.imemreq_q.enq.rdy
+    m.imemresp_en   //= s.imemresp_drop.out.en
+    m.imemresp_rdy  //= s.imemresp_drop.out.rdy
+
+    # dmem port
+    m.dmemreq_en    //= s.dmem.req.en
+    m.dmemreq_rdy   //= s.dmem.req.rdy
+    m.dmemreq_type  //= s.dmem.req.msg.type_
+    m.dmemresp_en   //= s.dmemresp_q.deq.en
+    m.dmemresp_rdy  //= s.dmemresp_q.deq.rdy
+
+    # xcel port
+    m.xcelreq_type  //= s.xcel.req.msg.type_
 
-    s.dpath = ProcDpath()(
+    m.xcelreq_en    //= s.xcel.req.en
+    m.xcelreq_rdy   //= s.xcel.req.rdy
+    m.xcelresp_en   //= s.xcelresp_q.deq.en
+    m.xcelresp_rdy  //= s.xcelresp_q.deq.rdy
+
+    # proc2mngr and mngr2proc
+    m.proc2mngr_en  //= s.proc2mngr.en
+    m.proc2mngr_rdy //= s.proc2mngr.rdy
+    m.mngr2proc_en  //= s.mngr2proc_q.deq.en
+    m.mngr2proc_rdy //= s.mngr2proc_q.deq.rdy
+
+    # commit inst for counting
+    m.commit_inst //= s.commit_inst
+
+    # Dpath
 
-      # imem ports
-      imemreq_addr  = s.imemreq_q.enq.msg.addr,
-      imemresp_data = s.imemresp_drop.out.ret,
+    s.dpath = m = ProcDpath()
 
-      # dmem ports
-      dmemreq_addr  = s.dmem.req.msg.addr,
-      dmemreq_data  = s.dmem.req.msg.data,
-      dmemresp_data = s.dmemresp_q.deq.ret.data,
+    # imem ports
+    m.imemreq_addr  //= s.imemreq_q.enq.msg.addr
```

Comment:
I imagine there could be a `stdlib.inspect_connect`  function that inspects two components and connect their ports based on the names... would be even better than SV `.*`

Reply:
connect_auto in pymtl2 basically ..